CMakeLists.txt

```diff
@@ -99,6 +99,25 @@ else ()
   set(OGRE_PLATFORM_X64 FALSE)
 endif ()
 
+# determine if strtoul_l, strtol_l, strtoull_l and strtoll_l are supported
+include(CheckFunctionExists)
+CHECK_FUNCTION_EXISTS(strtoul_l HAVE_STRTOUL_L)
```

Comment:
the result of the test must be stored in `OgreBuildSettings.h.in` to propagate to consuming projects.
Also, I would prefer musl to use the `#if OGRE_PLATFORM == OGRE_PLATFORM_ANDROID ...` fallback branch.
Here, you simplify the test to `#if OGRE_NO_LOCALE_STRCONVERT` and determine that variable as you currently do.

Reply:
Android fallback can't be used "as is" because this will raise a build failure due to locale_t being set to int